### ISSUE TITLE
Make geth fails in docker-compose for the l2geth service

### ISSUE THREAD

**Issue by qbzzt** [ethereum-optimism/optimism#1188]
**Bug description:**

When I run docker-compose I get this error:
```
(16/20) Installing pcre2 (10.36-r0)
(17/20) Installing git (2.32.0-r0)
(18/20) Installing linux-headers (5.10.41-r0)
(19/20) Installing make (4.3-r0)
(20/20) Installing musl-dev (1.2.2-r3)
Executing busybox-1.33.1-r2.trigger
OK: 147 MiB in 35 packages
Removing intermediate container e1d81c36656f
 ---> cc5ad963dee7
Step 3/10 : ADD ./l2geth /go-ethereum
 ---> 8df7636beeb5
Step 4/10 : RUN cd /go-ethereum && make geth
 ---> Running in 4bf7f610711f
env GO111MODULE=on go run build/ci.go install ./cmd/geth
make: env: Operation not permitted
make: *** [Makefile:16: geth] Error 127
ERROR: Service 'l2geth' failed to build: The command '/bin/sh -c cd /go-ethereum && make geth' returned a non-zero code: 2
```

**To Reproduce**
The exact steps are documented in https://github.com/qbzzt/optimism-tutorial/blob/main/README.md. You create a Debian 10 VM on GCP (it fails even with a 16 GB one), and then execute:
```
sudo apt install -y wget git docker docker.io build-essential docker-compose
curl -sL https://deb.nodesource.com/setup_12.x -o nodesource_setup.sh
sudo bash nodesource_setup.sh
sudo apt install -y nodejs
sudo npm install -g yarn hardhat
sudo usermod -a -G docker `whoami`
```

Then open a new command line and run:
```
git clone https://github.com/ethereum-optimism/optimism.git
cd optimism
yarn install
yarn build
cd ops
docker-compose build
```

**Expected behavior**
I expected `docker-compose` to create for me a working Optimism setup I can use to test things out.

**System Specs:**
 - OS: Debian 10
 - Package Version (or commit hash): Latest

**Comment by qbzzt** [comment]
I can run this on Ubuntu, so while it's still a bug, it's a much lower priority for me.

**Comment by K-Ho** [comment]
Synthetix's CI is running into this same bug. Pretty high priority to fix

**Comment by smartcontracts** [comment]
@K-Ho I think the solution is to upgrade Docker to version 20.10.0+

**Comment by smartcontracts** [comment]
@qbzzt can you still reproduce this on Debian today?

**Comment by nailkhaf** [comment]
> @K-Ho I think the solution is to upgrade Docker to version 20.10.0+

It works for me on pop-os 21.04 based on ubuntu 21.04.

[Stable docker channel](https://snapcraft.io/docker) 19.03.13 throws `make: env: Operation not permitted`, but beta 20.10.6 passes.
Meanwhile, ubuntu server 20.04 in Gnome Boxes works even with 19.03.13

**Comment by qbzzt** [comment]
> @qbzzt can you still reproduce this on Debian today?

No. :-) Good job.

I might change the tutorials back to Debian, simply because that's the GCP default.

**Comment by smartcontracts** [comment]
I believe this was fixed by #1199. Going to close. If someone is still experiencing this issue, please LMK and I will re-open.
